source/common/http/conn_manager_impl.cc

```diff
@@ -836,6 +847,18 @@ ConnectionManagerImpl::ActiveStream::requestHeadersForTags() const {
   return connection_manager_.config_.tracingConfig()->request_headers_for_tags_;
 }
 
+void ConnectionManagerImpl::ActiveStream::callHighWatermarkCallbacks() {
+  for (DownstreamWatermarkCallbacks* callbacks : watermark_callbacks_) {
+    callbacks->onAboveWriteBufferHighWatermark();
```

Comment:
Yeah, if we stuck with the filter callbacks it would be more obvious but also less performant.  

Meanwhile I think if the upstream connection dies, the router won't have a handle on upstream_request_ but Matt can sanity check me there.